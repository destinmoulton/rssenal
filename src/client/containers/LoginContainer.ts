import { connect } from "react-redux";

import LoginComponent, {
    ILoginComponentProps
} from "../components/LoginComponent";

import { loginUser } from "../redux/actions/auth.actions";

import * as Types from "../interfaces";

const mapStateToProps = (
    state: Types.IRootStoreState
): ILoginComponentProps => {
    return {
        authenticationError: state.auth.authenticationError
    };
};

const mapDispatchToProps = (
    dispatch: Types.IDispatch
): ILoginComponentProps => {
    return {
        loginUser: (username: string, password: string) => {
            return dispatch(loginUser(username, password));
        }
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(LoginComponent);
