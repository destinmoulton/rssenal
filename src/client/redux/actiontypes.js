
export const ENTRIES_GET_BEGIN = "ENTRIES_GET_BEGIN";
export const ENTRIES_GET_COMPLETE = "ENTRIES_GET_COMPLETE";
export const ENTRIES_UPDATE_BEGIN = "ENTRIES_UPDATE_BEGIN";
export const ENTRIES_UPDATE_COMPLETE = "ENTRIES_UPDATE_COMPLETE";

export const FEEDGROUPS_FETCHING = "FEEDGROUPS_FETCHING";
export const FEEDGROUPS_RECEIVED = "FEEDGROUPS_RECEIVED";
export const FEEDGROUPS_ADD_BEGIN = "FEEDGROUPS_ADD_BEGIN";
export const FEEDGROUPS_ADD_COMPLETE = "FEEDGROUPS_ADD_COMPLETE";
export const FEEDGROUPS_DELETE_BEGIN = "FEEDGROUPS_DELETE_BEGIN";
export const FEEDGROUPS_DELETE_COMPLETE = "FEEDGROUPS_DELETE_COMPLETE";
export const FEEDGROUPS_UPDATE_BEGIN = "FEEDGROUPS_UPDATE_BEGIN";
export const FEEDGROUPS_UPDATE_COMPLETE = "FEEDGROUPS_UPDATE_COMPLETE";

export const FEEDS_ADD_BEGIN = "FEEDS_ADD_BEGIN";
export const FEEDS_ADD_COMPLETE = "FEEDS_ADD_COMPLETE";
export const FEEDS_DELETE_BEGIN = "FEEDS_DELETE_BEGIN";
export const FEEDS_DELETE_COMPLETE = "FEEDS_DELETE_COMPLETE";
export const FEEDS_GETALL_COMPLETE = "FEEDS_GETALL_COMPLETE";
export const FEEDS_SETALL_UNREAD_COUNT = "FEEDS_SET_UNREAD_COUNT";
export const FEEDS_UPDATE_BEGIN = "FEEDS_UPDATE_BEGIN";
export const FEEDS_UPDATE_COMPLETE = "FEEDS_UPDATE_COMPLETE";

export const FILTER_CHANGE = "FILTER_CHANGE";

export const SETTINGS_CHANGE = "SETTINGS_CHANGE";