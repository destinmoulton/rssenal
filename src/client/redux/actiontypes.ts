export const AUTH_ERROR = "AUTH_ERROR";
export const AUTH_LOGOUT = "AUTH_LOGOUT";
export const AUTH_USER_IS_AUTHENTIC = "AUTH_USER_IS_AUTHENTIC";
export const AUTH_TOKEN_IS_VALIDATING = "AUTH_TOKEN_IS_VALIDATING";
export const AUTH_TOKEN_VALIDATION_COMPLETE = "AUTH_TOKEN_VALIDATION_COMPLETE";

export const ENTRIES_SET_ALL = "ENTRIES_SET_ALL";

export const FILTER_CHANGE = "FILTER_CHANGE";
export const FILTER_RESET = "FILTER_RESET";

export const FOLDERS_ADD_BEGIN = "FOLDERS_ADD_BEGIN";
export const FOLDERS_ADD_COMPLETE = "FOLDERS_ADD_COMPLETE";
export const FOLDERS_DELETE_BEGIN = "FOLDERS_DELETE_BEGIN";
export const FOLDERS_DELETE_COMPLETE = "FOLDERS_DELETE_COMPLETE";
export const FOLDERS_FETCHING = "FOLDERS_FETCHING";
export const FOLDERS_RECEIVED = "FOLDERS_RECEIVED";
export const FOLDERS_UPDATE_BEGIN = "FOLDERS_UPDATE_BEGIN";
export const FOLDERS_UPDATE_COMPLETE = "FOLDERS_UPDATE_COMPLETE";

export const FEEDS_ADD_BEGIN = "FEEDS_ADD_BEGIN";
export const FEEDS_ADD_COMPLETE = "FEEDS_ADD_COMPLETE";
export const FEEDS_DELETE_BEGIN = "FEEDS_DELETE_BEGIN";
export const FEEDS_DELETE_COMPLETE = "FEEDS_DELETE_COMPLETE";
export const FEEDS_GETALL_COMPLETE = "FEEDS_GETALL_COMPLETE";
export const FEEDS_SET_UNREAD = "FEEDS_SET_UNREAD";
export const FEEDS_CLEAR_UNREAD = "FEEDS_CLEAR_UNREAD";
export const FEEDS_UPDATE_BEGIN = "FEEDS_UPDATE_BEGIN";
export const FEEDS_UPDATE_COMPLETE = "FEEDS_UPDATE_COMPLETE";

export const MESSAGES_ADD = "MESSAGES_ADD";
export const MESSAGES_REMOVE = "MESSAGES_REMOVE";

export const SETTINGS_CHANGE = "SETTINGS_CHANGE";
