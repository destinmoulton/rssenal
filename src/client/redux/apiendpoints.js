export const API_BASE = "/api";

export const API_FEEDVALIDATION_BASE = API_BASE + "/validatefeedurl/";

export const API_FEEDGROUPS_BASE = API_BASE + "/feedgroups/";
export const API_FEEDGROUPS_GET_ALL = API_FEEDGROUPS_BASE;