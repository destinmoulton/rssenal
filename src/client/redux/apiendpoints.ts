export const API_BASE = "/api";

export const API_AUTH = API_BASE + "/auth/";

export const API_ENTRIES_BASE = API_BASE + "/entries/";

export const API_FEEDVALIDATION_BASE = API_BASE + "/validatefeedurl/";

export const API_FEEDS_BASE = API_BASE + "/feeds/";

export const API_FOLDERS_BASE = API_BASE + "/folders/";
export const API_FOLDERS_GET_ALL = API_FOLDERS_BASE;
