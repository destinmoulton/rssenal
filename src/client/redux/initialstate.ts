import { OrderedMap } from "immutable";
import * as Types from "../types";

export const AUTH_INITIAL_STATE: Types.IReducerStateAuth = {
    authenticationError: "",
    isAuthorized: false,
    isValidatingToken: false
};

export const ENTRIES_INITIAL_STATE: Types.IReducerStateEntries = {
    entries: OrderedMap<Types.TEntryID, Types.IEntry>()
};
