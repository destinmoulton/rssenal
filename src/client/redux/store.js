import { createStore, combineReducers, applyMiddleware } from 'redux'
import { routerReducer } from 'react-router-redux'
import thunk from 'redux-thunk';

import feedGroupsReducer from './reducers/feedgroups.reducer';

const store = createStore(
  combineReducers({
    feedGroups: feedGroupsReducer
  }),
  applyMiddleware(
    thunk
  )
);

export default store;
