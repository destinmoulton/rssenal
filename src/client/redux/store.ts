import { createStore, combineReducers, applyMiddleware } from "redux";

import { createLogger } from "redux-logger";
import thunk from "redux-thunk";

import authReducer from "./reducers/auth.reducer";
import entriesReducer from "./reducers/entries.reducer";
import filterReducer from "./reducers/filter.reducer";
import feedsReducer from "./reducers/feeds.reducer";
import foldersReducer from "./reducers/folders.reducer";
import messagesReducer from "./reducers/messages.reducer";
import settingsReducer from "./reducers/settings.reducer";

const logger = createLogger({
    collapsed: true
});
const store = createStore(
    combineReducers({
        authStore: authReducer,
        entriesStore: entriesReducer,
        filterStore: filterReducer,
        foldersStore: foldersReducer,
        feedsStore: feedsReducer,
        messagesStore: messagesReducer,
        settingsStore: settingsReducer
    }),
    applyMiddleware(logger, thunk)
);

export default store;
