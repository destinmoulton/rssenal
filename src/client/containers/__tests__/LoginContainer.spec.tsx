import * as React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { mount } from "enzyme";

import LoginContainer from "../LoginContainer";
import * as INIT_STATE from "../../redux/initialstate";

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

describe("<LoginContainer />", () => {
    it("renders and matches the snapshot", () => {
        const store = mockStore({
            authStore: INIT_STATE.AUTH_INITIAL_STATE
        });

        const wrapper = mount(
            <Provider store={store}>
                <LoginContainer />
            </Provider>
        );

        expect(wrapper).toMatchSnapshot();
    });
});
