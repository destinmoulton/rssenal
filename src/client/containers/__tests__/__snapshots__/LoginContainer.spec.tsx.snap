// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginContainer /> renders and matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(LoginComponent)>
    <LoginComponent
      authLoginUser={[Function]}
      authenticationError=""
    >
      <div
        className="login-form"
      >
        <style>
          
                body &gt; div,
                body &gt; div &gt; div,
                body &gt; div &gt; div &gt; div.login-form {
                    height: 100%;
                }
                
        </style>
        <Grid
          style={
            Object {
              "height": "100%",
            }
          }
          textAlign="center"
          verticalAlign="middle"
        >
          <div
            className="ui center aligned middle aligned grid"
            style={
              Object {
                "height": "100%",
              }
            }
          >
            <GridColumn
              style={
                Object {
                  "maxWidth": 450,
                }
              }
            >
              <div
                className="column"
                style={
                  Object {
                    "maxWidth": 450,
                  }
                }
              >
                <Header
                  as="h2"
                  className="rss-login-header"
                >
                  <h2
                    className="ui header rss-login-header"
                  >
                    <Icon
                      as="i"
                      name="rss"
                    >
                      <i
                        aria-hidden="true"
                        className="rss icon"
                      />
                    </Icon>
                      RSSenal
                  </h2>
                </Header>
                <Form
                  as="form"
                  size="large"
                >
                  <form
                    className="ui large form"
                    onSubmit={[Function]}
                  >
                    <Segment>
                      <div
                        className="ui segment"
                      >
                        <FormInput
                          as={[Function]}
                          control={[Function]}
                          fluid={true}
                          icon="user"
                          iconPosition="left"
                          name="username"
                          onChange={[Function]}
                          placeholder="Username"
                        >
                          <FormField
                            control={[Function]}
                            fluid={true}
                            icon="user"
                            iconPosition="left"
                            name="username"
                            onChange={[Function]}
                            placeholder="Username"
                          >
                            <div
                              className="field"
                            >
                              <Input
                                fluid={true}
                                icon="user"
                                iconPosition="left"
                                name="username"
                                onChange={[Function]}
                                placeholder="Username"
                                type="text"
                              >
                                <div
                                  className="ui fluid left icon input"
                                >
                                  <input
                                    name="username"
                                    onChange={[Function]}
                                    placeholder="Username"
                                    type="text"
                                  />
                                  <Icon
                                    as="i"
                                    name="user"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="user icon"
                                    />
                                  </Icon>
                                </div>
                              </Input>
                            </div>
                          </FormField>
                        </FormInput>
                        <FormInput
                          as={[Function]}
                          control={[Function]}
                          fluid={true}
                          icon="lock"
                          iconPosition="left"
                          name="password"
                          onChange={[Function]}
                          placeholder="Password"
                          type="password"
                        >
                          <FormField
                            control={[Function]}
                            fluid={true}
                            icon="lock"
                            iconPosition="left"
                            name="password"
                            onChange={[Function]}
                            placeholder="Password"
                            type="password"
                          >
                            <div
                              className="field"
                            >
                              <Input
                                fluid={true}
                                icon="lock"
                                iconPosition="left"
                                name="password"
                                onChange={[Function]}
                                placeholder="Password"
                                type="password"
                              >
                                <div
                                  className="ui fluid left icon input"
                                >
                                  <input
                                    name="password"
                                    onChange={[Function]}
                                    placeholder="Password"
                                    type="password"
                                  />
                                  <Icon
                                    as="i"
                                    name="lock"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="lock icon"
                                    />
                                  </Icon>
                                </div>
                              </Input>
                            </div>
                          </FormField>
                        </FormInput>
                        <Button
                          as="button"
                          className="rss-login-button"
                          fluid={true}
                          onClick={[Function]}
                          role="button"
                          size="large"
                        >
                          <button
                            className="ui large fluid button rss-login-button"
                            onClick={[Function]}
                            role="button"
                          >
                            Login
                          </button>
                        </Button>
                      </div>
                    </Segment>
                  </form>
                </Form>
              </div>
            </GridColumn>
          </div>
        </Grid>
      </div>
    </LoginComponent>
  </Connect(LoginComponent)>
</Provider>
`;
